packages/gatsby-image/src/index.js

```diff
@@ -37,18 +37,17 @@ const convertProps = props => {
     convertedProps.loading = `eager`
   }
 
+  // convert fluid & fixed to arrays so we only have to work with arrays
+  convertedProps.fluid = [].concat(convertedProps.fluid).filter(Boolean)
```

Comment:
@brimtown did [something similar to this originally in `convertProps()`](https://github.com/gatsbyjs/gatsby/pull/13395/files/46dc7318d291b299bd79b3d94bf3542e5ab5612b#diff-5941a73c5d794b120294a0429b6351c1R16) and [I wasn't too fond of it](https://github.com/gatsbyjs/gatsby/pull/13395#discussion_r277495092). Especially now that the method indicates it's intended for handling deprecation transitions. Which I guess can apply here and encourage users to always wrap in an array(or have the related sharp plugin/transformer accommodate).

@brimtown 's version assigned the first element if an array were present, while the code has now changed to expect an array to work with regardless. I would say, an earlier suggestion I made would be more appropriate here(nitpicking), as it's less operations, and a little more clearer to read at a glance what the desired outcome is:

```JS
convertedProps.fluid = Array.isArray(fluid) ? fluid : [fluid]
convertedProps.fixed = Array.isArray(fixed) ? fixed : [fixed]
```

Reply:
Disregard this suggestion. As it is no longer in scope of just fluid or fixed, it can return a `[undefined]` value that'd mess things up.

Reply:
I'll make it more verbose with wrapping it. I don't think `[].concat` is super hard to understand
```
// convert fluid & fixed to arrays so we only have to work with arrays
if (convertedProps.fluid) {
  convertedProps.fluid = [].concat(convertedProps.fluid)
}
if (convertedProps.fixed) {
  convertedProps.fixed = [].concat(convertedProps.fixed)
}
```